File: 19 - Remove Nth Node From End of List/main.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode RemoveNthFromEnd(ListNode head, int n) {
        int node_count = 0;
        var temp = head;
        ListNode prev = null;

        while(temp is not null){
            node_count += 1;
            temp=temp.next;
        }

        if (node_count == 1){
            return null;
        }

        temp = head;
        int cur_count = 0;

        while(temp is not null){
            cur_count += 1;
            if (cur_count == node_count-(n-1)){
                if (prev is null){
                    return temp.next;
                }
                else{
                    prev.next = temp.next;
                    return head;
                }
            }
            prev = temp;
            temp = temp.next;
        }

        return head;
    }
}

File: 19 - Remove Nth Node From End of List/main.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        node_l = []
        temp = head
        while(temp):
            node_l.append(temp)
            temp = temp.next
        if(len(node_l) == 1):
            return None
        if(n < len(node_l) and n != 1):
            node_l[-n-1].next = node_l[-n+1]
        elif(n == len(node_l)):
            head = node_l[1]
        else:
            node_l[-2].next = None
            
        return head